File: leet_1572_diagonal_sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        r = 0
        L = len(mat)
        for i in range(L):
            r += mat[i][i] + mat[i][L - i - 1]
        if L % 2 == 1:
            r -= mat[L // 2][L // 2]
        return r

File: leet_1572_diagonal_sum.go
func diagonalSum(mat [][]int) int {
    i := 0
    r := 0
    len := len(mat)
    for i < len{
        r += mat[i][i] + mat[i][len - i - 1]
        i++
    }
    if len % 2 == 1 {
        r -= mat[len / 2][len / 2]
    }
    return r
}
